group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -408,6 +408,31 @@ public MetadataImage image() {
         return metadataImage;
     }
 
+    /**
+     *
+     * @param groupId
+     * @param createIfNotExists
+     * @return
+     * @throws GroupIdNotFoundException
+     */
+    public Group getOrMaybeCreateSimpleGroup(
```

Comment:
This naming is a bit confusing since I thought we only get the simple group. (Not any group)

We should probably will in line 415 and that may help clear it up a bit.

Reply:
Yeah, I agree with you. Let me rework this part.

Reply:
I have reworked this code. Let me know what you think?